From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Thu, 29 May 2025 20:10:02 -0500
Subject: [PATCH] Fix 0.7 regression breaking builds
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This fixes builds that fail with the below error:
│B│ slibtool-shared: error: --mode must be specified.
│B│ g-ir-scanner: warning: Option --strip-prefix has been deprecated;
│B│ linking of temporary binary failed: Command '['slibtool-shared', '--prefer-sltdl', '', '--mode=link', '--tag=CC', '--silent', 'clang', '-o', '/mason/build/x86_64/avahi.git/avahi-gobject/tmp-introspectjnsjuan0/Avahi-0.6', '-export-dynamic', '-D_FORTIFY_SOURCE=2', '-O2', '-Werror=odr', '-Werror=strict-aliasing', '-ffat-lto-objects', '-flto=thin', '-fno-omit-frame-pointer', '-mno-omit-leaf-frame-pointer', '-fstack-protector', '--param', 'ssp-buffer-size=32', '-g', '-fasynchronous-unwind-tables', '-march=x86-64-v2', '-mtune=ivybridge', '-mprefer-vector-width=128', '-pipe', '-Wformat', '-Wformat-security', '-Wno-error', '-fPIC', '-fstack-protector', '-std=c99', '-Wall', '-W', '-Wextra', '-pedantic', '-pipe', '-Wformat', '-Wold-style-definition', '-Wdeclaration-after-statement', '-Wfloat-equal', '-Wmissing-declarations', '-Wmissing-prototypes', '-Wstrict-prototypes', '-Wredundant-decls', '-Wmissing-noreturn', '-Wshadow', '-Wendif-labels', '-Wpointer-arith', '-Wbad-function-cast', '-Wcast-qual', '-Wcast-align', '-Wwrite-strings', '-fdiagnostics-show-option', '-Wno-cast-qual', '-fno-strict-aliasing', '/mason/build/x86_64/avahi.git/avahi-gobject/tmp-introspectjnsjuan0/Avahi-0.6.o', '-L.', '-Wl,--no-as-needed', 'libavahi-gobject.la', '-lgio-2.0', '-lgobject-2.0', '-Wl,--export-dynamic', '-lgmodule-2.0', '-pthread', '-lglib-2.0', '-Wl,--build-id=sha1', '-Wl,--compress-debug-sections=zstd', '-Wl,--icf=safe', '-Wl,-Bsymbolic-functions', '-Wl,-O2,--gc-sections', '-Wl,-z,pack-relative-relocs', '-Wl,-z,relro,-z,now', '-flto=thin']' returned non-zero exit status 2.
│B│ see --identifier-prefix and --symbol-prefix.

Caused by SLIBTOOL_FLAGS being undefined and introducing an extra space into the command which breaks argument parsing.
---
 m4/slibtool.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/m4/slibtool.m4 b/m4/slibtool.m4
index 944ba0c..6a37f9d 100644
--- a/m4/slibtool.m4
+++ b/m4/slibtool.m4
@@ -543,7 +543,7 @@ _SLIBTOOL_ARG_WITH([sysroot],[absolute path to the target's sysroot],[slibtool_s
 # --------------------
 slibtool_set_flavor
 LIBTOOL='$(SLIBTOOL)'
-SLIBTOOL="${SLIBTOOL} ${SLIBTOOL_FLAGS}"
+SLIBTOOL="\$(strip ${SLIBTOOL} ${SLIBTOOL_FLAGS})"
 
 AC_SUBST([LIBTOOL])
 AC_SUBST([SLIBTOOL])
